contentcuration/contentcuration/static/js/bundle_modules/base.js

```diff
@@ -1,13 +1,45 @@
 var attachfastclick = require("fastclick");
 var $ = require("jquery");
+var get_cookie = require("utils/get_cookie");
 global.$ = $;
 global.jQuery = $;
+require("handlebars/helpers");
 
-require("bootstrap/less/bootstrap.less");
-require("bootstrap/dist/js/npm.js");
-require("bootstrap/dist/css/bootstrap.css");
 require("../../less/styles.less");
-require("../../less/global-variables.less");
+require("bootstrap/dist/js/npm.js");
+
+var csrftoken = get_cookie("csrftoken") || "";
+
+function csrfSafeMethod(method) {
+    // these HTTP methods do not require CSRF protection
+    return (/^(GET|HEAD|OPTIONS|TRACE)$/.test(method));
+}
+
+function sameOrigin(url) {
+    // test that a given url is a same-origin URL
+    // url could be relative or scheme relative or absolute
+    var host = document.location.host; // host + port
+    var protocol = document.location.protocol;
+    var sr_origin = '//' + host;
+    var origin = protocol + sr_origin;
+    // Allow absolute or scheme relative URLs to same origin
+    return (url == origin || url.slice(0, origin.length + 1) == origin + '/') ||
+        (url == sr_origin || url.slice(0, sr_origin.length + 1) == sr_origin + '/') ||
+        // or any other URL that isn't scheme relative or absolute i.e relative.
+        !(/^(\/\/|http:|https:).*/.test(url));
+}
+
+$.ajaxSetup({
+    cache: false,
+    beforeSend: function(xhr, settings) {
+		if (!csrfSafeMethod(settings.type) && sameOrigin(settings.url)) {
```

Comment:
Not a suggestion, but a question to make sure I understand: you only include the CSRF token for certain methods when the request's url is the same origin as the site's. This is to prevent the CSRF token from being leaked to a malicious server, if they could somehow make a request?